src/python/pants/backend/python/register.py

```diff
@@ -83,6 +84,8 @@ def rules():
         *pipenv_requirements.rules(),
         *poetry_requirements.rules(),
         *python_requirements.rules(),
+        # Test
+        *debug_goals.rules(),
```

Comment:
These should probably go in their own backend / `register.py`, similar to the JVM and Go debug goals: https://github.com/pantsbuild/pants/tree/main/src/python/pants/backend/experimental/java/debug_goals

That's mostly because the goal names / flags are experimental / unstable: it would be amazing to eventually merge them.